### ISSUE TITLE
Will there be support for an auto generated references.d.ts with correct order of TS-files when dealing with dependencies?

### ISSUE THREAD

**Issue by yngvark** [TypeStrong/grunt-ts#133]
I have not actually been using grunt-ts, so I apologize if I have misunderstood how it works when writing this. Anyway:

I am using the example from the main documentation from "JavaScript generation and ordering". Just to recap: Let's say your are using the "out" option in config to create a concatened JS-file containing all compiled JS-code. Also, you have for instance a base class that needs to loaded before its subclasses in the generated JS-file. The current way to make sure that the base class JS-code is inserted above the subclasses' code in the generated JS-file, is to manually order the TS-files in references.d.ts.

For large projects, this seems cumbersome. This is also a common problem, and part of what RequireJS solves, but AMD/RequireJS support has been deprecated.

In other words, it would be great if grunt-ts was smart enough to create a output JS-file where the compiled TS-files are inserted in correct order, according to their dependency graph. I suppose it also would be good if the generated references.d.ts also had the same, auto generated, correct order of its TS-files.
